config/routes.rb

```diff
@@ -1,11 +1,13 @@
 Rails.application.routes.draw do
 
+
   devise_for :users
   resources :users, only: [:update]
    
    resources :topics do
-     resources :posts, except: [:index]
-   end
+   resources :posts, except: [:index]
+   resources :comments, only: [:create]
```

Comment:
comments should be nested under posts, and posts should be nested under topics. This is why you have your error I guess.